x-pack/plugins/osquery/public/results/attachments/external_reference.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ExternalReferenceAttachmentType } from '@kbn/cases-plugin/public/client/attachment_framework/types';
+import { lazy } from 'react';
+// import { EuiButtonIcon } from '@elastic/eui';
+
+const AttachmentContentLazy = lazy(() => import('./external_references_content'));
+
+// TODO waiting for Metadata to add "add to timeline" in here
+// const AttachmentActions: React.FC = () => (
+//   <EuiButtonIcon
+//     data-test-subj="test-attachment-action"
+//     onClick={() => {}}
+//     iconType="arrowRight"
+//     aria-label="See attachment"
+//   />
+// );
+
+export const getExternalReferenceAttachmentRegular: () => ExternalReferenceAttachmentType = () => ({
+  id: 'osquery',
+  icon: 'logoOsquery',
+  displayName: 'Osquery',
+  getAttachmentViewObject: () => ({
+    type: 'regular',
+    event: 'attached Osquery results',
+    timelineIcon: 'logoOsquery',
+    // actions: <AttachmentActions />,
+    // @ts-expect-error update types
```

Comment:
What's the error that we're receiving here?

Reply:
```TS2322: Type 'LazyExoticComponent<(props: AttachmentContentProps) => Element>' is not assignable to type 'LazyExoticComponent<FC<ExternalReferenceAttachmentViewProps>> | undefined'.   Type 'LazyExoticComponent<(props: AttachmentContentProps) => Element>' is not assignable to type 'LazyExoticComponent<FC<ExternalReferenceAttachmentViewProps>>'.     Type 'LazyExoticComponent<(props: AttachmentContentProps) => Element>' is not assignable to type 'ExoticComponent<ExternalReferenceAttachmentViewProps & { children?: ReactNode; }>'.       Types of parameters 'props' and 'props' are incompatible.         Type 'ExternalReferenceAttachmentViewProps & { children?: ReactNode; }' is not assignable to type 'AttachmentContentProps'.           Types of property 'externalReferenceMetadata' are incompatible.             Type '{ [x: string]: JsonValue; } | null' is not assignable to type '{ actionId: string; agentIds: string[]; queryId: string; }'.               Type 'null' is not assignable to type '{ actionId: string; agentIds: string[]; queryId: string; }'.  types.ts(20, 3): The expected type comes from property 'children' which is declared here on type 'AttachmentViewObject<ExternalReferenceAttachmentViewProps>'```